lib/compress.js

```diff
@@ -2130,9 +2130,6 @@ merge(Compressor.prototype, {
                     extract_declarations_from_unreachable_code(compressor, self.body, a);
                     return make_node(AST_BlockStatement, self, { body: a });
                 }
-            } else {
-                // self instanceof AST_Do
-                return self.body;
```

Comment:
How would `} else { return self; }` alter the flow? The following `if (self instanceof AST_While)` would not be triggered and then `return self;` would be called.